osp-workshop-2024.sh

```diff
@@ -60,14 +66,38 @@ there is proper $kubeconfig_file provided and that at least one OpenStack DataPl
 EOF
             exit 1
         fi
-        # Now lets remove from the inventory file lines about ssh key which
-        # is mounted in the ansible runner pod but we want to use key provided
-        # with the -p parameter instead
-        sed -i '/ansible_ssh_private_key_file: /d' $inventory_file
+       # Now lets build very simple inventory file with just required options
+       python3 prepare_inventory.py $tmp_edpm_inventory_file $inventory_file
+    fi
+}
+
+function configure_ssh_jump_host() {
+    grep -q $jump_host $SSH_CONFIG
+    if [ $? -eq 0 ]; then
+        return
+    fi
+    for hostname in $(grep ansible_host $inventory_file | awk -F':' '{print $2}'); do
+        cat <<EOF >>$SSH_CONFIG
+Host $hostname
+    ProxyJump $jump_host
+EOF
+    done
+}
+
+function ensure_private_key_exists() {
+    local key_file=$1
+    if [ ! -e $key_file ]; then
+        $oc_bin --kubeconfig $kubeconfig_file -n $oc_namespace extract secret $DATAPLANE_SSH_KEY_SECRET_NAME --keys=ssh-privatekey --to=$key_file
+    fi
+    # Add private key to the ssh-agent
+    pidof ssh-agent
+    if [ $? -ne 0 ]; then
+        eval $(ssh-agent)
     fi
+    ssh-add $key_file
 }
 
-while getopts "b:dKs:i:p:c:o:n" opt_key; do
+while getopts "b:dKs:i:p:c:o:n:j:" opt_key; do
```

Comment:
Is this fixing the `n` parameter intentionally?

Reply:
actually no, I didn't notice issue with -n until now, but yes, it will fix it. Do you think I should move this fix to separate commit?

Reply:
it's fine :) I just wondered